driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -74,6 +74,78 @@ public MappingManager(Session session, ProtocolVersion protocolVersion) {
         // which nodes might join the cluster later.
         // At least if protocol >=2 we know there won't be any 1.2 nodes ever.
         this.isCassandraV1 = (protocolVersion == ProtocolVersion.V1);
+        session.getCluster().register(new SchemaChangeListenerBase() {
+
+            @Override
+            public void onTableRemoved(TableMetadata table) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (mapper.getTableMetadata().equals(table)) {
+                            LOGGER.error("Table {} has been removed; existing mappers for @Entity annotated {} will not work anymore", table.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onTableChanged(TableMetadata current, TableMetadata previous) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (mapper.getTableMetadata().equals(previous)) {
+                            LOGGER.warn("Table {} has been altered; existing mappers for @Entity annotated {} might not work properly anymore",
+                                    previous.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onUserTypeRemoved(UserType type) {
+                synchronized (udtCodecs) {
+                    Iterator<MappedUDTCodec<?>> it = udtCodecs.values().iterator();
+                    while (it.hasNext()) {
+                        MappedUDTCodec<?> codec = it.next();
+                        if (type.equals(codec.getCqlType())) {
+                            LOGGER.error("User type {} has been removed; existing mappers for @UDT annotated {} will not work anymore",
+                                    type, codec.getUdtClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onUserTypeChanged(UserType current, UserType previous) {
```

Comment:
Wondering, I don't think what I'm about to suggest is necessary, but just curious.

If a UDT is changed, shouldn't we also in theory clear out any `Mapper` that has a field that maps to that type?  Alternatively we could update the underlying `ColumnMapper` for that field.  I don't think it's necessary as if the change is compatible it'll keep working, if not it'll stop working and there's no way to fix it (other than updating your code).